msm/msm

```diff
@@ -23,114 +23,166 @@
+        skill=$(echo "$skills" | grep -i ".*:.*${iskill}.*")
+        if [[ ! -z "${exact_match}" ]]; then
+            skill=${exact_match}
+        fi
+        git_line=$(echo "$skill" | sed 's/\:.*//')
+    
+        if [[ "${skill}" ==  *$'\n'* ]]; then
+            echo -e "Your search has multiple choices\n\n$skill" | sed 's/.*://g'
+            return 3
+        else
+            if [[ -z "${git_line}" ]]; then
+                echo "A ${iskill} skill was not found"
+                return 3
+            fi
+        repo_line=$(($git_line + 2))
+        repo=$(echo "$skill_list" | sed -n $repo_line'{p;q;}' | sed 's/[[:space:]]//g' | sed 's/url=//g')
         fi 
-        echo "Skill installed!"
-}
+    fi
+    git_name=$(echo "${repo}" | sed 's/.*\///')
+    name=$(echo "$git_name" | sed 's/.git//')
+    echo "Cloning repository"
+    git clone "${repo}" >> /dev/null
+    if ! cd "${name}" ; then
+        echo "Unable to access directory ${name}!"
+        return 102
+    fi
+    if [[ -f "requirements.txt" ]]; then
+        echo "Installing libraries requirements"
+        if [[ $picroft -eq 1 ]]; then
+            if workon mycroft ; then
+                if ! pip install -r requirements.txt ; then
+                    echo "Unable to install requirements for skill ${iskill}!"
+                    deactivate
+                    return 121
+                fi
+            else
+                echo "Unable to activate mycroft virtualenv!"
+                deactivate
+                return 120
+            fi
+        else 
+            if ! sudo pip install -r requirements.txt ; then
+                echo "Unable to install requirements for skill ${iskill}!"
+                return 121
+            fi
+        fi
+        
+    fi 
+    echo "The ${iskill} skill has been installed!"
+done
 
-function update() {
-        goto_skills_dir
-	for d in $(ls -d */); do
-		if git -C "$d" rev-parse --git-dir > /dev/null 2>&1; then
-			cd $d
-			if [[ -z $(git status --porcelain) ]]; then
-				git fetch
-				git reset --hard origin/master 
-			fi
-			cd ..
-		fi
-	done
 }
 
-function install_defaults() {
-	skills=( "alarm" "audio-record" "configuration" "date-time" "desktop-launcher" "ip" "joke" "hello-world" "media" "npr-news" "naptime" "pairing" "personal" "reminder" "installer" "singing" "speak" "spelling" "stop" "stock" "volume" "weather" "wiki" "wolfram-alpha" "mark1-demo" )
-	for i in "${skills[@]}"
-	do
-		if [ ! -d "$mycroft_skill_folder/skill-$i" ]; then
-			install "" "https://github.com/MycroftAI/skill-$i.git"	
-		fi
-	done
-	update
-	echo "Installed!"
-	exit 0
+function update() {
+cd "${mycroft_skill_folder}"
+for d in $(find "${mycroft_skill_folder}" -mindepth 1 -maxdepth 1 -type d |grep -v '.git'$ ); do
+    if git -C "$d" rev-parse --git-dir > /dev/null 2>&1; then
+        cd "${d}" 
+        if [[ -z $(git status --porcelain) ]]; then
+        git fetch
+        git reset --hard origin/master 
+    fi
+fi
+done
```

Comment:
I believe so from a read of the google style guide. I've also been using 4 instead of two spaces...oops.